lib/apiservers/engine/backends/container.go

```diff
@@ -219,53 +219,74 @@ func (c *Container) TaskInspect(cid, cname, eid string) (*models.TaskInspectResp
 
 }
 
+func (c *Container) TaskWait(cid, cname, eid string) error {
+	// obtain a portlayer client
+	client := c.containerProxy.Client()
+
+	handle, err := c.Handle(cid, cname)
+	if err != nil {
+		return err
+	}
+
+	// wait the Task to start
+	config := &models.TaskWaitConfig{
+		Handle: handle,
+		ID:     eid,
+	}
+
+	params := tasks.NewWaitParamsWithContext(ctx).WithConfig(config)
+	_, err = client.Tasks.Wait(params)
+	if err != nil {
+		switch err := err.(type) {
+		case *tasks.WaitInternalServerError:
+			return InternalServerError(err.Payload.Message)
+		default:
+			return InternalServerError(err.Error())
+		}
+	}
+
+	return nil
+}
+
 // ContainerExecCreate sets up an exec in a running container.
 func (c *Container) ContainerExecCreate(name string, config *types.ExecConfig) (string, error) {
 	defer trace.End(trace.Begin(name))
 
-	if !config.Detach {
-		return "", fmt.Errorf("%s only supports detached exec commands at this time", ProductName())
-	}
-
 	// Look up the container name in the metadata cache to get long ID
 	vc := cache.ContainerCache().GetContainer(name)
 	if vc == nil {
 		return "", NotFoundError(name)
 	}
 	id := vc.ContainerID
 
-	handle, err := c.Handle(id, name)
+	// Is it running?
+	state, err := c.containerProxy.State(vc)
 	if err != nil {
 		return "", InternalServerError(err.Error())
 	}
+	if state.Restarting {
```

Comment:
Can the actual check be moved to the PL, then do error mapping for exact docker errors here?